logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/auth/ClearUserDataUseCase.kt

```diff
@@ -0,0 +1,51 @@
+package com.wire.kalium.logic.feature.auth
+
+import com.wire.kalium.logic.AuthenticatedDataSourceSet
+import com.wire.kalium.logic.data.client.ClientRepository
+import com.wire.kalium.logic.data.client.MLSClientProvider
+import com.wire.kalium.logic.data.id.QualifiedID
+import com.wire.kalium.logic.functional.isLeft
+import com.wire.kalium.logic.kaliumLogger
+
+interface ClearUserDataUseCase {
+    suspend operator fun invoke()
+}
+
+class ClearUserDataUseCaseImpl @Suppress("LongParameterList") constructor(
+    private val userId: QualifiedID,
+    private val authenticatedDataSourceSet: AuthenticatedDataSourceSet,
+    private val clientRepository: ClientRepository,
+    private val mlsClientProvider: MLSClientProvider
+) : ClearUserDataUseCase {
+
+    override suspend operator fun invoke() {
+        clearCrypto()
```

Comment:
I suggest to validate the user session here, like if it's invalid then clear it! that would be safer.

Reply:
The use case is private now so it's safer 😄 We shouldn't clear the session there because it's used in the logout so we don't want to remove the session but to to keep it as `Invalid`.